Source/DafnyCore/Options/DisableNonLinearArithmeticOption.cs

```diff
@@ -0,0 +1,15 @@
+namespace Microsoft.Dafny;
+
+class DisableNonLinearArithmeticOption : BooleanOption {
+  public static readonly DisableNonLinearArithmeticOption Instance = new();
+  public override string LongName => "disable-nonlinear-arithmetic";
+
+  public override string Description => @"      
+Reduce Dafny's knowledge of non-linear arithmetic (*,/,%).
+  
+Results in more manual work, but also produces more predictable behavior.".TrimStart();
+  public override string PostProcess(DafnyOptions options) {
+    options.DisableNLarith = Get(options);
```

Comment:
Isn't this option deprecated?  I thought we wanted to remove it rather than port it over.

Reply:
Judging by `https://github.com/dafny-lang/libraries/tree/master/src/NonlinearArithmetic` I thought it was in active use. Is it not?

Reply:
It's in use, but we're not super happy about that:

```

  /noNLarith
      Reduce Z3's knowledge of non-linear arithmetic (*,/,%).
  
      Results in more manual work, but also produces more predictable
      behavior. (This switch will perhaps be replaced by /arith in the
      future. For now, it takes precedence of /arith.)
```

Reply:
I've marked it as experimental